### PR TITLE
[ALLUXIO-2895] Support Multiple Arguments with WildCards for Alluxio Shell Commands	

### PR DESCRIPTION
**Pull request by lijiang2**
https://alluxio.atlassian.net/browse/ALLUXIO-2895

### CODE REVIEW

**Comment by lijiang2** [comment]
* Implemented validateArgs for each individual command. Multiple wildcard argument is allowed for following commands: chmod, count, free, load, ls, pin, rm, unpin.
* In WithWildCardPathCommand, allowed the code to take more than one argument, instead of only args[0].

File: shell/src/main/java/alluxio/shell/command/CheckConsistencyCommand.java [hunk 1]
```diff
@@ -135,4 +135,15 @@ public String getDescription() {
         + "under storage will be returned. An administrator should then reconcile the differences."
         + "Specify -r to repair the inconsistent files.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
+    boolean valid = args.length <= getNumOfArgs();
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CheckConsistencyCommand.java]: Based on my suggestion, we don't need to modify `validateArgs` here and leave it to inherit from super class

File: shell/src/main/java/alluxio/shell/command/AbstractShellCommand.java [hunk 2]
```diff
@@ -49,14 +49,7 @@ protected AbstractShellCommand(FileSystem fs) {
    * @param args the arguments for the command, excluding the command name and options
    * @return whether the args are valid
    */
-  protected boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/AbstractShellCommand.java]: After read the code again, I think there is a simpler way to accomplish this JIRA with less change introduced: 
- we can simply leave the this `AbstractShellCommand#validataArgs` unchanged.
- For subclass of `AbstractShellCommand` that we want to take non-fixed number of args, we could simply override their `validateArgs` to be the following, like what you did in `CatCommand`:
```java
@Override
public boolean validateArgs(String... args) { return args.length >= 1; }
```

File: shell/src/main/java/alluxio/shell/command/ChecksumCommand.java [hunk 3]
```diff
@@ -87,4 +87,15 @@ public String getUsage() {
   public String getDescription() {
     return "Calculates the md5 checksum of a file in the Alluxio filesystem.";
   }
+
+  @Override
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/ChecksumCommand.java]: ditto, and the other classes.

File: shell/src/main/java/alluxio/shell/command/ChgrpCommand.java [hunk 4]
```diff
@@ -95,4 +95,15 @@ public String getDescription() {
     return "Changes the group of a file or directory specified by args."
         + " Specify -R to change the group recursively.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/ChgrpCommand.java]: Let's remove this and still requires `chgrp` to take 2 arguments

File: shell/src/main/java/alluxio/shell/command/CatCommand.java [hunk 5]
```diff
@@ -72,4 +72,7 @@ public String getUsage() {
   public String getDescription() {
     return "Prints the file's contents to the console.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) { return true; }
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CatCommand.java]: `validateArgs` for this command should return `true` when `args.length >= 1`

File: shell/src/main/java/alluxio/shell/command/ChmodCommand.java [hunk 6]
```diff
@@ -101,4 +101,7 @@ public String getDescription() {
     return "Changes the permission of a file or directory specified by args."
         + " Specify -R to change the permission recursively.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) { return true; }
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/ChmodCommand.java]: Let's remove this and still requires `chmod` to take 2 arguments

File: shell/src/main/java/alluxio/shell/command/ChownCommand.java [hunk 7]
```diff
@@ -95,4 +95,15 @@ public String getDescription() {
     return "Changes the owner of a file or directory specified by args."
         + " Specify -R to change the owner recursively.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/ChownCommand.java]: Let's remove this and still requires `chown` to take 2 arguments

File: shell/src/main/java/alluxio/shell/command/CopyFromLocalCommand.java [hunk 8]
```diff
@@ -66,4 +66,15 @@ public String getUsage() {
   public String getDescription() {
     return "Copies a file or a directory from local filesystem to Alluxio filesystem.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CopyFromLocalCommand.java]: Remove this and keep it unchanged

File: shell/src/main/java/alluxio/shell/command/CopyToLocalCommand.java [hunk 9]
```diff
@@ -65,4 +65,15 @@ public String getUsage() {
   public String getDescription() {
     return "Copies a file or a directory from the Alluxio filesystem to the local filesystem.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CopyToLocalCommand.java]: Remove and keep this unchanged.

File: shell/src/main/java/alluxio/shell/command/CountCommand.java [hunk 10]
```diff
@@ -91,4 +91,7 @@ public String getUsage() {
   public String getDescription() {
     return "Displays the number of files and directories matching the specified prefix.";
   }
+
+  @Override
+  public boolean validateArgs(String... args) { return true; }
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CountCommand.java]: Remove and keep this unchanged.

File: shell/src/main/java/alluxio/shell/command/CpCommand.java [hunk 11]
```diff
@@ -602,4 +602,15 @@ private static boolean isAlluxio(String scheme) {
   private static boolean isFile(String scheme) {
     return "file".equals(scheme);
   }
+
+  @Override
+  public boolean validateArgs(String... args) {
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CpCommand.java]: Remove and keep this unchanged.

File: shell/src/main/java/alluxio/shell/command/AbstractShellCommand.java [hunk 12]
```diff
@@ -53,7 +53,7 @@ protected boolean validateArgs(String... args) {
     boolean valid = args.length == getNumOfArgs();
     if (!valid) {
       System.out.println(getCommandName() + " takes " + getNumOfArgs() + " arguments, " + " not "
-          + args.length + "\n");
+              + args.length + "\n");
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/AbstractShellCommand.java]: unnecessary change

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java [hunk 13]
```diff
@@ -286,8 +286,14 @@ public MountTResponse mount(final String alluxioPath, final String ufsPath,
     return RpcUtils.callAndLog(LOG, new RpcCallableThrowsIOException<MountTResponse>() {
       @Override
       public MountTResponse call() throws AlluxioException, IOException {
+//        if (numOfArgs == 0) {
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java]: can you remove these commented lines? @lijiang2

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java [hunk 14]
```diff
@@ -286,8 +286,14 @@ public MountTResponse mount(final String alluxioPath, final String ufsPath,
...
+//          mFileSystemMaster.getMountTable();
+//        } else {
+//          mFileSystemMaster.mount(new AlluxioURI(alluxioPath), new AlluxioURI(ufsPath),
+//              new MountOptions(options));
+//        }
         mFileSystemMaster.mount(new AlluxioURI(alluxioPath), new AlluxioURI(ufsPath),
-            new MountOptions(options));
+                new MountOptions(options));
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java]: indentation is still off

File: shell/src/main/java/alluxio/shell/command/MountCommand.java [hunk 15]
```diff
@@ -112,4 +112,8 @@ public String getUsage() {
   public String getDescription() {
     return "Mounts a UFS path onto an Alluxio path.";
   }
+  @Override
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/MountCommand.java]: needs one extra line before line 115 to pass style checker
@lijiang2

**Comment by apc999** [comment]
/home/jenkins/workspace/Alluxio-Pull-Request-Builder/shell/src/main/java/alluxio/shell/command/MountCommand.java:115: error: 'METHOD_DEF' should be separated from previous statement.
